github/resource_github_team_repository.go

```diff
@@ -36,7 +36,6 @@ func resourceGithubTeamRepository() *schema.Resource {
 				Type:         schema.TypeString,
 				Optional:     true,
 				Default:      "pull",
-				ValidateFunc: validateValueFunc([]string{"pull", "triage", "push", "maintain", "admin"}),
```

Comment:
Is there a way to pull all possible roles from GitHub and then validate against that?

Reply:
I am sure we could use [this API](https://docs.github.com/en/rest/orgs/custom-roles) for that. 

Do you know of an example where something similar is done before? Or could you suggest where I should create the function. First time contributor :)

Reply:
Seems like we do some other validation in `utils.go` (e.g. validateValueFunc and validateTeamIDFunc) but I don't see anyone calling back to github api here.

Reply:
I think we probably need to wait for this: https://github.com/google/go-github/pull/2336? But I am also a little out of my depth here.

Reply:
The go-github functionality for merged a little while ago.

Reply:
It's merged but unfortunately has not been released yet; the last release of the library was March 1st. We'll have to wait for their next official release to use the functionality in [go-github#2336](https://github.com/google/go-github/pull/2336).

@joshuahancox is correct and thus far we've chosen to hard-code those validation functions rather than use extra API calls. I'm a little hesitant to change that approach here, especially as we tend to see API overuse on more intensive Terraform runs.

Reply:
https://github.com/integrations/terraform-provider-github/pull/1138 will bump `go-github` to include the necessary functionality.

Reply:
I've just updated and merged #1138!